upup/pkg/fi/cloudup/terraform/target_hcl2.go

```diff
@@ -132,23 +133,34 @@ func (t *TerraformTarget) writeProviders(buf *bytes.Buffer) {
 		Write(buf, 0, fmt.Sprintf("provider %q", providerName))
 	buf.WriteString("\n")
 
-	// Add the second provider definition for managed files
-	if t.filesProvider != nil {
+	// Add any additional provider definition for managed files
+	keys := keysForMap(t.TerraformWriter.Providers)
+	sort.Strings(keys)
+	for _, key := range keys {
+		filesProvider := t.TerraformWriter.Providers[key]
 		providerBody := map[string]string{}
 		providerBody["alias"] = "files"
-		for k, v := range t.filesProvider.Arguments {
+		for k, v := range filesProvider.Arguments {
 			providerBody[k] = v
 		}
 		for k, v := range tfGetFilesProviderExtraConfig(t.clusterSpecTarget) {
 			providerBody[k] = v
 		}
 		mapToElement(providerBody).
 			ToObject().
-			Write(buf, 0, fmt.Sprintf("provider %q", t.filesProvider.Name))
+			Write(buf, 0, fmt.Sprintf("provider %q", filesProvider.Name))
 		buf.WriteString("\n")
 	}
 }
 
+func keysForMap[K comparable, V any](m map[K]V) []K {
+	var keys []K
+	for k := range m {
+		keys = append(keys, k)
+	}
+	return keys
```

Comment:
Sort here instead of in the caller? `K` is always `string`.

Reply:
Sure... the challenge with generics seems to be when to stop, but this is a clear win and it's comparable anyway...